tool/update_version.dart

```diff
@@ -284,43 +327,48 @@ class AutoUpdateCommand extends Command {
   AutoUpdateCommand() {
     argParser.addOption('type',
         abbr: 't',
-        allowed: ['dev', 'dev,patch', 'dev,major', 'patch', 'minor', 'major'],
+        allowed: ['release', 'dev', 'patch', 'minor', 'major'],
         allowedHelp: {
+          'release': 'strips any pre-release versions from the version.',
           'dev':
-              'bumps the version to the next dev pre-release value (minor by default)',
-          'dev,patch': 'bumps the version to the next dev pre-patch value',
-          'dev,major': 'bumps the version to the next dev pre-major value',
-          'patch': 'bumps the version to the next patch value',
-          'minor': 'bumps the version to the next minor value',
-          'major': 'bumps the version to the next major value',
+              'bumps the version to the next dev pre-release value (minor by default).',
+          'patch':
+              'bumps the version to the next patch value, and sets the dev version to 0.',
+          'minor':
+              'bumps the version to the next minor value, and sets the dev version to 0.',
+          'major':
+              'bumps the version to the next major value, and sets the dev version to 0.',
         },
         mandatory: true,
         help: 'Bumps the devtools version by the selected type.');
   }
 
   @override
-  void run() {
+  void run() async {
     final type = argResults!['type'].toString();
     final currentVersion = versionFromPubspecFile();
     String? newVersion;
     if (currentVersion == null) {
       throw 'Could not automatically determine current version.';
     }
     switch (type) {
-      case 'dev':
-        newVersion = incrementDevVersion(currentVersion, 'minor');
+      case 'release':
+        newVersion = stripPreReleases(currentVersion);
         break;
-      case 'dev,patch':
-        newVersion = incrementDevVersion(currentVersion, 'patch');
-        break;
-      case 'dev,major':
-        newVersion = incrementDevVersion(currentVersion, 'major');
+      case 'dev':
+        newVersion = incrementDevVersion(currentVersion);
         break;
       default:
         newVersion = incrementVersionByType(currentVersion, type);
-    }
-    if (newVersion == null) {
-      throw 'Failed to determine the newVersion.';
+
+        // Doing a proper version update so cycle the release notes
+        await resetReleaseNotes(
```

Comment:
I don't think we want to do release notes for patch updates for example, only once a month for our major or minor release